rflx/model/session.py

```diff
@@ -96,15 +96,30 @@ def exception_transition(self) -> Optional[Transition]:
     def actions(self) -> Sequence[stmt.Statement]:
         return self.__actions
 
+    @property
+    def is_null(self) -> bool:
+        return not self.__transitions
+
     @property
     def has_exceptions(self) -> bool:
         return any(
-            isinstance(a, (stmt.Append, stmt.Extend))
+            isinstance(a, (stmt.Append, stmt.Extend, stmt.Write))
             or (
                 isinstance(a, stmt.Assignment)
-                and isinstance(
-                    a.expression,
-                    (expr.Selected, expr.Head, expr.Comprehension, expr.MessageAggregate),
+                and (
+                    a.expression.findall(
+                        lambda x: isinstance(
+                            x,
+                            (
+                                expr.Selected,
```

Comment:
Even a message with just one path can have fields which are detected as invalid during parsing. For example, a field value could be out of range. This could be also the case for `Config` in the PIN spec. As we cannot infer that `Config'Valid` was already checked before, accessing `Config.Max_Retries` would lead to an exception transition if `Config.Max_Retries` is not in `0 .. 6`.

There are certainly cases where exception transitions would not be needed (e.g., a message with one path and only modular integers), but I don't think it is worth to add special handling for them, as exception transitions will not be used for `Selected` in the long term (cf. #569).

Reply:
I missed the other opportunities for a message to be invalid. Let's forget about that idea then and rather do #569.